Framework/Algorithms/test/RebinTest.h

```diff
@@ -118,6 +118,59 @@ class RebinTest : public CxxTest::TestSuite {
     AnalysisDataService::Instance().remove("test_out");
   }
 
+  void test_make_three_params() {
+    // test that if only a binwidth is given, will create an xmin, xmax from xstep
+    Workspace2D_sptr test_in1D = Create1DWorkspace(50);
+    AnalysisDataService::Instance().add("test_in1D", test_in1D);
+
+    Rebin rebin;
+    rebin.initialize();
+    rebin.setPropertyValue("InputWorkspace", "test_in1D");
+    rebin.setPropertyValue("OutputWorkspace", "test_out");
+    rebin.setPropertyValue("Params", "2.0");
+    TS_ASSERT_THROWS_NOTHING(rebin.execute());
+    TS_ASSERT(rebin.isExecuted());
+
+    // ensure that the properties are not changed in default mode
+    std::vector<double> rbParams = rebin.getProperty("Params");
+    TS_ASSERT_EQUALS(rbParams.size(), 1);
```

Comment:
isn't this testing that rbParams is of size 1, not 3?

Reply:
There are two things tested here.  The first set of asserts ensures that the default algorithm will create a full set of rebin params if only given a step size.  This is verified by the algorithm being able to execute without errors.  This was previously untested behavior.

The second set of asserts verifies that the "Params" property was not changed.  This is the original behavior of this algo, and needs to remain preserved.

Reply:
I think I am just a bit confused (if I'm reading it correctly) by the differing behaviour between this and e.g. L165, where Params is still being given a single value, but constructs the xmin and xmax from the data, sorry I should have been clearer with the comment

Reply:
This part is marked outdated, but the tests still exist in L53 and L970.  In Default mode, the `exec` method should fill in the endpoints of the rebin parameters using the workspace endpoints and use them internally, but not change the property.  In every other binning mode, the `validateInputs` method should fill in the endpoints and change the property before `exec` runs.  I think it's more clean in the refactor since your comments.